### PR TITLE
update azure tasks

### PR DESCRIPTION
**Pull request by yaananth**

### CODE REVIEW

**Comment by yaananth** [comment]
#567

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 [hunk 1]
```diff
@@ -200,12 +208,29 @@ $servicePackageFile = Find-Files -SearchPattern "$CsPkg"
...
+
+   
 if (!$azureService)
-{
-    Write-Host "New-AzureService -ServiceName $ServiceName -Location $ServiceLocation"
-    $azureService = New-AzureService -ServiceName $ServiceName -Location $ServiceLocation
+{    
+    $azureService = "New-AzureService -ServiceName `"$ServiceName`""
+    if($NewServiceAffinityGroup) {
+        $azureService += " -AffinityGroup `"$NewServiceAffinityGroup`""
```
Comment (ericsciple) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: Do you need to escape special characters within NewServiceAffinityGroup?

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 [hunk 2]
```diff
@@ -200,12 +208,29 @@ $servicePackageFile = Find-Files -SearchPattern "$CsPkg"
...
+    }
+    elseif($ServiceLocation) {
+         $azureService += " -Location `"$ServiceLocation`""
```
Comment (ericsciple) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: Do you need to escape special characters within ServiceLocation?

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 [hunk 3]
```diff
@@ -200,12 +208,29 @@ $servicePackageFile = Find-Files -SearchPattern "$CsPkg"
...
+    }
+    else {
+        Write-Error "Either AffinityGroup or ServiceLocation must be specified"
```
Comment (ericsciple) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: It looks like you might want to throw instead.

File: Tasks/AzureCloudPowerShellDeployment/task.json [hunk 4]
```diff
@@ -45,11 +52,14 @@
     },
     {
       "name": "ServiceLocation",
-      "type": "string",
+      "type": "pickList",
```
Comment (ericsciple) [inline Tasks/AzureCloudPowerShellDeployment/task.json]: Does picklist allow overriding if the value isn't in the list?

File: Tasks/AzureCloudPowerShellDeployment/task.json [hunk 5]
```diff
@@ -97,6 +107,27 @@
...
+      "type": "string",
+      "label": "Additional Arguments",
+      "defaultValue": "",
+      "required": false,
+      "helpMarkDown": "Pass in additional arguments while creating a brand new service. These will be passed on to `New-AzureService` commandlet. Eg: `-Label 'MyTestService'`",
+      "groupName": "newServiceAdvancedOptions"
+    },
+    {
+      "name": "NewServiceAffinityGroup",
+      "type": "pickList",
```
Comment (ericsciple) [inline Tasks/AzureCloudPowerShellDeployment/task.json]: Where do the values get loaded?

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 (hunk 1 above)
Comment (yaananth) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: usually Affinity groups shouldn't have any special characters, , I think it's fine if it blows away

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 (hunk 2 above)
Comment (yaananth) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: usually ServiceLocation shouldn't have any special characters, , I think it's fine if it blows away

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 (hunk 3 above)
Comment (yaananth) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: yeah, i will update

File: Tasks/AzureCloudPowerShellDeployment/task.json (hunk 4 above)
Comment (yaananth) [inline Tasks/AzureCloudPowerShellDeployment/task.json]: Yep, that's correct

File: Tasks/AzureCloudPowerShellDeployment/task.json (hunk 5 above)
Comment (yaananth) [inline Tasks/AzureCloudPowerShellDeployment/task.json]: in the UI, we added new source binding, so that they are dynamically loaded

**Pull request merged**